src/lazy/encoder/text/v1_1/value_writer.rs

```diff
@@ -0,0 +1,286 @@
+        })
+    }
+
+    fn sexp_writer(self) -> IonResult<Self::SExpWriter> {
+        Ok(TextSExpWriter_1_1 {
+            writer_1_0: self.value_writer_1_0.sexp_writer()?,
+        })
+    }
+
+    fn struct_writer(self) -> IonResult<Self::StructWriter> {
+        Ok(TextStructWriter_1_1 {
+            writer_1_0: self.value_writer_1_0.struct_writer()?,
+        })
+    }
+
+    fn eexp_writer<'a>(self, _macro_id: impl Into<MacroIdRef<'a>>) -> IonResult<Self::EExpWriter> {
+        IonResult::encoding_error("e-expressions cannot have annotations")
+    }
+}
+
+pub struct TextListWriter_1_1<'value, W: Write> {
+    writer_1_0: TextListWriter_1_0<'value, W>,
+}
+
+impl<'value, W: Write> MakeValueWriter for TextListWriter_1_1<'value, W> {
+    type ValueWriter<'a> = TextValueWriter_1_1<'a, W> where Self: 'a;
+
+    fn make_value_writer(&mut self) -> Self::ValueWriter<'_> {
+        TextValueWriter_1_1 {
+            value_writer_1_0: self.writer_1_0.make_value_writer(),
+        }
+    }
+}
+
+impl<'value, W: Write> SequenceWriter for TextListWriter_1_1<'value, W> {
+    type Resources = ();
+
+    fn close(self) -> IonResult<Self::Resources> {
+        self.writer_1_0.close()
+    }
+}
+
+pub struct TextSExpWriter_1_1<'value, W: Write> {
+    writer_1_0: TextSExpWriter_1_0<'value, W>,
+}
+
+impl<'value, W: Write> MakeValueWriter for TextSExpWriter_1_1<'value, W> {
+    type ValueWriter<'a> = TextValueWriter_1_1<'a, W> where Self: 'a;
+
+    fn make_value_writer(&mut self) -> Self::ValueWriter<'_> {
+        TextValueWriter_1_1 {
+            value_writer_1_0: self.writer_1_0.make_value_writer(),
+        }
+    }
+}
+
+impl<'value, W: Write> SequenceWriter for TextSExpWriter_1_1<'value, W> {
+    type Resources = ();
+
+    fn close(self) -> IonResult<Self::Resources> {
+        self.writer_1_0.close()
+    }
+}
+
+pub struct TextStructWriter_1_1<'value, W: Write> {
+    writer_1_0: TextStructWriter_1_0<'value, W>,
+}
+
+impl<'value, W: Write> FieldEncoder for TextStructWriter_1_1<'value, W> {
+    fn encode_field_name(&mut self, name: impl AsRawSymbolTokenRef) -> IonResult<()> {
+        self.writer_1_0.encode_field_name(name)
+    }
+}
+
+impl<'value, W: Write> MakeValueWriter for TextStructWriter_1_1<'value, W> {
+    type ValueWriter<'a> = TextValueWriter_1_1<'a, W> where Self: 'a;
+
+    fn make_value_writer(&mut self) -> Self::ValueWriter<'_> {
+        TextValueWriter_1_1 {
+            value_writer_1_0: self.writer_1_0.make_value_writer(),
+        }
+    }
+}
+
+impl<'value, W: Write> StructWriter for TextStructWriter_1_1<'value, W> {
+    fn close(self) -> IonResult<()> {
+        self.writer_1_0.close()
+    }
+}
+
+pub struct TextEExpWriter_1_1<'value, W: Write> {
+    // There is no e-exp writer in 1.0 to which we can delegate,
+    // but we can re-use the TextContainerWriter_1_0 for a lot of the formatting.
+    container_writer: TextContainerWriter_1_0<'value, W>,
+}
+
+impl<'value, W: Write> TextEExpWriter_1_1<'value, W> {
```

Comment:
Does `*EExpWriter` need a way to write an expression group?

Reply:
It will eventually; expression groups are a `TODO` in all readers/writers.